numpy/lib/arraypad.py

```diff
@@ -797,9 +798,22 @@ def pad(array, pad_width, mode='constant', **kwargs):
         # Create a new zero padded array
         padded, _ = _pad_simple(array, pad_width, fill_value=0)
         # And apply along each axis
-        for axis in range(padded.ndim):
-            np.apply_along_axis(
-                function, axis, padded, pad_width[axis], axis, kwargs)
+
+        in_dims = list(range(array.ndim))
```

Comment:
Why is this needed? `axis in range` does the job.

Reply:
Leftover from before moveaxis was being used, good catch

Reply:
yes, thanks for catching that. I fixed it and then rebased this to a single commit.